files/new_user_api_notify/ostapi-newuserticket.php

```diff
@@ -0,0 +1,97 @@
+    'key'=>'_OSTAPIKEY_'  // API Key goes here
+);
+# NOTE: some people have reported having to use "http://your.domain.tld/api/http.php/tickets.json" instead.
+
+if($config['url'] === 'http://your.domain.tld/api/tickets.json') {
+    echo "<p style=\"color:red;\"><b>Error: No URL</b><br>You have not configured this script with your URL!</p>";
+    echo "Please edit this file ".__FILE__." and add your URL at line 18.</p>";
+    die();  
+}
+
+if(IsNullOrEmptyString($config['key']) || ($config['key'] === 'PUTyourAPIkeyHERE'))  {
+    echo "<p style=\"color:red;\"><b>Error: No API Key</b><br>You have not configured this script with an API Key!</p>";
+    echo "<p>Please log into osticket as an admin and navigate to: Admin panel -> Manage -> Api Keys then add a new API Key.<br>";
+    echo "Once you have your key edit this file ".__FILE__." and add the key at line 19.</p>";
+    die();
+}
+
+# Fill in the data for the new ticket, this will likely come from $_POST.
+# NOTE: your variable names in osT are case sensiTive. 
+# So when adding custom lists or fields make sure you use the same case
+
+$message = '<html><body style="font-family: Helvetica, Arial, san-serif; font-size:12pt;"><p>';
+$message .= 'Someone just created a new _ENVIRNAME_ account that needs approval.<br>';
+$message .= 'The following new account was created:<br><br>';
+$message .= '__username__ created an account at __time__ GMT from the following IP address: __ip__<br><br> ';
+$message .= 'The account is currently disabled, please confirm the account justification with the account manager, enable the account, and send the new user the appropriate notification.<br><br>';
+$message .= 'Note - Ticket created via API from PWM.</p>';
+
+$data = array(
+    'name'      =>      'The PWM via API',  // from name aka User/Client Name
+    'email'     =>      '_EMAILFROMPWM_',  // from email aka User/Client Email
+    'subject'   =>      'NEW USER: __username__ created an account in _ENVIRNAME_',  // test subject, aka Issue Summary
+    'message'   =>      "data:text/html;charset=utf-8,$message",
+    'topicId'   =>      '1', // the help Topic that you want to use for the ticket 
+    'attachments' => array()
+);
+
+# more fields are available and are documented at:
+# https://github.com/osTicket/osTicket-1.8/blob/develop/setup/doc/api/tickets.md
+
+if($debug=='1') {
+    print_r($data);
+    die();
+}
+
+#pre-checks
+function_exists('curl_version') or die('CURL support required');
+function_exists('json_encode') or die('JSON support required');
+
+#set timeout
+set_time_limit(30);
+
+#curl post
+$ch = curl_init();
+curl_setopt($ch, CURLOPT_URL, $config['url']);
+curl_setopt($ch, CURLOPT_POST, 1);
+curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
+curl_setopt($ch, CURLOPT_USERAGENT, 'osTicket API Client v1.8');
+curl_setopt($ch, CURLOPT_HEADER, FALSE);
+curl_setopt($ch, CURLOPT_HTTPHEADER, array( 'Expect:', 'X-API-Key: '.$config['key']));
```

Comment:
I am not sure which part of this you are looking to have templated by Salt

Reply:
Well I should rephrase. I could understand which parts you would potentially do but from a quick glance, it would entail changing another script that is doing the substitution.